book/security.rst

```diff
@@ -0,0 +1,660 @@
+
+       class ProductController extends Controller
+       {
+           /**
+            * @AclAncestor("product_edit")
+            */
+           public function editAction()
+           {
+               // ...
+           }
+       }
+
+   Sometimes you want to protect a controller method coming from code that you do not control.
+   Therefore, you cannot add the ``@AclAncestor`` annotation to it. Use the ``bindings`` key in the
+   YAML configuration of your ACL to define which method(s) should be protected:
+
+   .. code-block:: yaml
+
+       # src/Acme/DemoBundle/Resources/config/acl.yml
+       product_edit:
+           type: entity
+           class: AcmeDemoBundle:Product
+           permission: EDIT
+           bindings:
+               - class: Acme\DemoBundle\Controller\ProductController
+                 method: editAction
+
+Using Param Converters
+~~~~~~~~~~~~~~~~~~~~~~
+
+When using the `@ParamConverter annotation`_ from the SensioFrameworkExtraBundle, the behavior is a
+bit different. Besides just checking the ``@Acl`` annotations, it is also checked if the Doctrine
+query performed to convert a request parameter affects entity whose class name was described in the
+``@Acl`` configuration. In this case, access will be checked for the queried object. Otherwise the
+check is performed in the entity's class scope.
+
+Data Grids
+~~~~~~~~~~
+
+Records that are part of a data grid are automatically protected by the OroSecurityBundle. View
+permissions are attached to each record of a data grid.
+
+Protecting Custom DQL Queries
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When building custom Doctrine DQL queries, you may want to reduce the result set being returned to
+the set of domain objects the user is granted access to. To achieve this, use the ACL helper
+provided by the OroSecurityBundle::
+
+    // src/Acme/DemoBundle/Controller/DemoController.php
+    namespace Acme\DemoBundle\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+
+    class DemoController extends Controller
+    {
+        public function protectedAction()
+        {
+            $repository = $this->getDoctrine()->getRepository('AcmeDemoBundle:Product');
+            $queryBuilder = $repository
+                ->createQueryBuilder('p');
+                ->where('p.price > :price')
+                ->orderBy('p.price', 'ASC')
+                ->setParameter('price', 19.99);
+            $aclHelper = $this->get('oro_security.acl_helper');
+            $query = $aclHelper->apply($queryBuilder, 'VIEW');
+
+            // ...
+        }
+    }
+
+In this example, first a query is built which selects all products from the database which are more
+expensive than ``19.99`` order by their price. Then, the query builder is passed to the ``apply()``
+method of the ``oro_security.acl_helper`` service which wraps the query builder in a proxy. This
+proxy then filters the results for the current user based on the given permission.
```

Comment:
We does not wraps the query in a proxy. We just add Oro\Bundle\SecurityBundle\ORM\Walker\AclHelper:ORO_ACL_WALKER hint into the query object, and then query will be run, it modify by Oro\Bundle\SecurityBundle\ORM\Walker\AclWalker class with data from ORO_ACL_WALKER hint data